src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeployedToEnvironmentNotifierStep.java

```diff
@@ -113,12 +98,13 @@ public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher,
                 return;
             }
 
+            BitbucketDeploymentEnvironment environment = getEnvironment();
+            if (environment == null) {
+                listener.error("Notify Bitbucket Server of deployment: could not send notification as the step is incorrectly configured");
```

Comment:
message can be polished a bit, but you get the gist of it. We just accept an invalid config and if it is invalid we do nothing.